### PR TITLE
Rename module to `celo-blockchain`

### LINKED ISSUE #1344: Rename our packages to not be called github.com/ethereum/go-ethereum/*
## Description

In `go.mod`, rename our package to:

```
module github.com/celo-org/celo-blockchain
```

And update all the imports which say `github.com/ethereum/go-ethereum` to say `github.com/celo-org/celo-blockchain`.

Additionally, we should have some protection against accidentally importing anything from go-ethereum due to new imports added as part of a merge.  This could be part of CI, or maybe part of the build.  The key point is to ensure we are not pulling in go-ethereum packages.  Probably grepping `go.mod` would be sufficient, but this needs to be verified.

Or, as suggested by @trianglesphere, there may be a way to automate the renaming of imports using gofmt.  See https://spf13.com/post/go-fmt/

## Context

The current naming, besides being inaccurate, prevents third-party projects from adding both go-ethereum and celo-blockchain as dependencies if they need to.

### PR DESCRIPTION
**Pull request by trianglesphere**
### Description

This changes the module name from ethereum/go-ethereum to celo-org/celo-blockchain. The main motivation behind that change is opening a possibility for seamless integration of celo-blockchain with projects already depending on the original go-ethereum module. Specifically, a project with a single dependency can do a single replace in their go.mod file but a project already having a dependency to ethereum/go-ethereum can not import Celo without this change.

This PR was initially opened by @lukasz-zimnoch of keep. I made some adjustments to docs and merged in master.

### Tested

N/A

### Related issues

- Fixes #1344

### Backwards compatibility

External contributors will need to change their import path. Some temporary hiccups while developers move everything over.

### CODE REVIEW

File: interfaces.go [hunk 1]
```diff
@@ -22,8 +22,8 @@ import (
 	"errors"
 	"math/big"
```
Comment (lukasz-zimnoch) [inline interfaces.go:24]: The main package will be still `ethereum`. Can we also rename it to `celo` (line 18)?

Comment (trianglesphere) [reply interfaces.go:24]: Yep

Comment (oneeman) [reply interfaces.go:24]: I'm less enthusiastic about this change.  The resulting conflicts in code when merging form upstream will not be trivial like ones from imports (and code changes more often than imports).  Would leaving the alias as ethereum cause any problems?  After all, any other project that imports from celo-blockchain can alias it to celo themselves.

Comment (trianglesphere) [reply interfaces.go:24]: Ok, we're renaming it externally as `celo`, but keeping it as `ethereum` internally. Practically the internal rename should have no effect on other users.

File: interfaces.go [hunk 2]
```diff
@@ -15,15 +15,15 @@
 // along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
 
 // Package ethereum defines interfaces for interacting with Ethereum.
```
Comment (nategraf) [inline interfaces.go:17]:
```suggestion
// Package celo defines interfaces for interacting with Celo.
```

**Review by nategraf — APPROVED** [review]
When I saw this PR, I thought to myself: "Ah! The perfect PR to hide a backdoor! No one would ever expect a reviewer to look at every file!" And so I did! Ha! No exploits....

I did find one place where we should replace "Ethereum" with "Celo", so theres that

**Pull request merged**